src/main/java/io/aiven/kafka/auth/VerdictCache.java

```diff
@@ -20,40 +20,66 @@
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import org.apache.kafka.common.security.auth.KafkaPrincipal;
 
-import io.aiven.kafka.auth.json.AivenAcl;
+import io.aiven.kafka.auth.acl.AivenAcl;
+import io.aiven.kafka.auth.acl.AllowAcl;
+import io.aiven.kafka.auth.acl.DenyAcl;
+import io.aiven.kafka.auth.acl.LoadedAcls;
 
 public class VerdictCache {
-    private final List<AivenAcl> aclEntries;
+    private final List<AllowAcl> allowAclEntries;
+    private final List<DenyAcl> denyAclEntries;
     private final Map<String, Boolean> cache = new ConcurrentHashMap<>();
 
-    private VerdictCache(final List<AivenAcl> aclEntries) {
-        this.aclEntries = aclEntries;
+    private VerdictCache(final List<DenyAcl> denyAclEntries, final List<AllowAcl> allowAclEntries) {
+        this.denyAclEntries = denyAclEntries;
```

Comment:
We can validate here that values cannot be null:

```suggestion
        this.denyAclEntries = Objects.nonNull(denyAclEntries);
```